File: Leetcode/1334/1334.go
func findTheCity(n int, edges [][]int, distanceThreshold int) int {
	const INF = 0x3f3f3f3f
	am := make([][]int, n)
	for i := 0; i < n; i++ {
		am[i] = make([]int, n)
	}
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			am[i][j] = INF
		}
	}
	for i := 0; i < n; i++ {
		am[i][i] = 0
	}
	var u, v, w int
	for i := 0; i < len(edges); i++ {
		u = edges[i][0]
		v = edges[i][1]
		w = edges[i][2]
		am[u][v] = w
		am[v][u] = w
	}
	for k := 0; k < n; k++ {
		for i := 0; i < n; i++ {
			for j := 0; j < n; j++ {
				if am[i][j] > am[i][k] + am[k][j] {
					am[i][j] = am[i][k] + am[k][j]
				}
			}
		}
	}
	stats := make([]int, n)
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			if i != j && am[i][j] <= distanceThreshold {
				stats[i]++
			}
		}
	}
	minVal := INF
	for i := 0; i < n; i++ {
		if stats[i] < minVal {
			minVal = stats[i]
		}
	}
	for i := n-1; i >= 0; i-- {
		if stats[i] == minVal {
			return i
		}
	}
	return -1
}

File: Leetcode/1334/1334.cpp
constexpr int INF = 0x3f3f3f3f;
constexpr int MAXN = 105;

int am[MAXN][MAXN];
int stats[MAXN];

class Solution {
public:
    int findTheCity(int n, vector<vector<int>>& edges, int distanceThreshold) {
        // Floyd-Warshall: O(n^3)
        memset(am, 0x3f, sizeof am);
        for (int i = 0; i < n; ++i) {
            am[i][i] = 0;
        }
        int u,v,w;
        for (const auto &e : edges) {
            u = e[0];
            v = e[1];
            w = e[2];
            am[u][v] = w;
            am[v][u] = w;
        }
        for (int k = 0; k < n; ++k) {
            for (int i = 0; i < n; ++i) {
                for (int j = 0; j < n; ++j) {
                    // am[i][j] = min(am[i][j], am[i][k]+am[k][j])
                    am[i][j] = min(am[i][j], am[i][k]+am[k][j]);
                }
            }
        }
        memset(stats, 0, sizeof stats);
        for (int i = 0; i < n; ++i) {
            for (int j = 0; j < n; ++j) {
                if (i != j && am[i][j] <= distanceThreshold) {
                    ++stats[i];
                }
            }
        }
        int minReachableCities = INF;
        for (int i = 0; i < n; ++i) {
            minReachableCities = min(minReachableCities, stats[i]);
        }
        for (int i = n - 1; i >= 0; --i) {
            if (stats[i] == minReachableCities) {
                return i;
            }
        }
        return -1;
    }
};